src/libraries/System.Runtime/tests/System/Reflection/NullabilityInfoContextTests.cs

```diff
@@ -984,6 +991,90 @@ public void TestValueTupleGenericTypeParameters(string fieldName, NullabilitySta
             Assert.Equal(param1, tupleInfo.GenericTypeArguments[0].ReadState);
             Assert.Equal(param2, tupleInfo.GenericTypeArguments[1].ReadState);
         }
+
+        public static IEnumerable<object?[]> GenericInheritanceTestData()
+        {
+            yield return new object?[] { typeof(ListOfUnconstrained<string?>), NullabilityState.NotNull, null };
+            yield return new object?[] { typeof(ListUnconstrainedOfNullable<string>), NullabilityState.NotNull, null };
+            yield return new object?[] { typeof(ListUnconstrainedOfNullableOfObject<>), NullabilityState.NotNull, null };
+            yield return new object?[] { typeof(ListOfArrayOfNullableString), NullabilityState.NotNull, NullabilityState.Nullable };
+            yield return new object?[] { typeof(ListOfNotNull<ListOfNotNull<object>>), NullabilityState.NotNull, NullabilityState.NotNull };
+            yield return new object?[] { typeof(ListOfListOfObject<string?>), NullabilityState.NotNull, NullabilityState.NotNull };
+            yield return new object?[] { typeof(ListMultiGenericOfNotNull<object?, string, string?>), NullabilityState.NotNull, null };
```

Comment:
Agreed. In this case I meant to have:

```
public class ListMultiGenericOfNotNull<T, U, V> : List<U>  // currently : List<V>
	where T : class? 
	where U : class 
	where V : class? 
{ 
}
```

So this will be NotNull again once I change that.